table.lua

```diff
@@ -303,19 +311,39 @@ function flib_table.filter(tbl, filter, array_insert)
 end
 
 --- Retrieve the value at the key, or insert the default value.
---- @param table table
+--- @param tbl table
 --- @param key any
 --- @param default_value any
 --- @return any
-function flib_table.get_or_insert(table, key, default_value)
-  local value = table[key]
+function flib_table.get_or_insert(tbl, key, default_value)
+  local value = tbl[key]
   if not value then
-    table[key] = default_value
+    tbl[key] = default_value
     return default_value
   end
   return value
 end
 
+--- Inserts element value at position pos in list if element does not exist.
```

Comment:
This function is a good idea. However, it would be waaaay more performant to create a builder function for a `UniqueArray` object, then use a metatable to check a hash table to see if the element exists yet.

Taking this from Recipe Book's code:
```lua
function util.unique_string_array(initial_tbl)
  initial_tbl = initial_tbl or {}
  local hash = {}
  for _, value in pairs(initial_tbl) do
    hash[value] = true
  end
  return setmetatable(initial_tbl, {
    __newindex = function(tbl, key, value)
      if not hash[value] then
        hash[value] = true
        rawset(tbl, key, value)
      end
    end,
  })
end
```
Barring that, it should at least be noted that this function is O(n).